feat: Update base images and configurations in Dockerfile

Itxaka opened the pull request:
- Changed `kairos-init` image from `v0.5.17` to `v0.6.0-RC1`
- Updated Fedora base image from `40` to `43`
- Updated Rocky Linux base image from `9` to `10`
- Updated Ubuntu base images to `20.04`, `22.04`, and `24.04`
- Updated Alpine version from `3.21` to `3.22`
- Updated OpenSUSE version from `15.6` to `16`
- Updated Debian version from `12` to `13`

These changes ensure compatibility with the latest versions of the base images and improve overall stability and security.

<!-- please add an icon to the title of this PR (see https://github.com/kairos-io/kairos/blob/master/CONTRIBUTING.md#step-5-push-your-feature-branch-to-your-fork), and delete this line and similar ones -->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kairos-io/kairos/issues/3736

Itxaka committed:
feat: Update base images and configurations in Dockerfile

- Changed `kairos-init` image from `v0.5.17` to `v0.6.0-RC1`
- Updated Fedora base image from `40` to `43`
- Updated Rocky Linux base image from `9` to `10`
- Updated Ubuntu base images to `20.04`, `22.04`, and `24.04`
- Updated Alpine version from `3.21` to `3.22`
- Updated OpenSUSE version from `15.6` to `16`
- Updated Debian version from `12` to `13`

These changes ensure compatibility with the latest versions of the base images and improve overall stability and security.

Signed-off-by: Itxaka <itxaka@kairos.io>

Itxaka committed:
feat: Update base images in Dockerfile and configuration files
- Changed base image from `rockylinux:10` to `rockylinux/rockylinux:10` for consistency and clarity
- Updated references in `image-master.yaml`, `release.yaml`, and `release-arm.yaml` to reflect the new base image

Signed-off-by: Itxaka <itxaka@kairos.io>

Itxaka committed:
feat: Update build configuration for example jobs
- Set `fail-fast` to false in the job strategy to allow all jobs to run even if one fails

Signed-off-by: Itxaka <itxaka@kairos.io>

mauromorales commented on .github/workflows/release-arm.yaml:
```diff
@@ -117,16 +117,16 @@ jobs:
         kubernetes_version: ${{ fromJson(needs.get-k3s-versions.outputs.kubernetes_versions) }}
         base_image:
-          - "opensuse/leap:15.6"
+          - "opensuse/leap:26"
```
```suggestion
          - "opensuse/leap:16"
```

mauromorales commented:
LOL I thought it was only one distro which was old, maybe when we moved the pipelines we forgot to have the renovate syntax? anywyas, it makes me wonder what are our users actually using, sounds like Ubuntu is the main candidate :p

Itxaka committed:
Update .github/workflows/release-arm.yaml

Co-authored-by: Mauro Morales <mauro.morales@spectrocloud.com>
Signed-off-by: Itxaka <itxaka@kairos.io>

Itxaka committed:
Merge branch 'master' into update_base_images

Signed-off-by: Itxaka <itxaka@kairos.io>

Itxaka committed:
feat: Update base images and release matchers

- Changed base image for OpenSUSE from `opensuse/leap:15.6` to `opensuse/leap:16`
- Updated release matcher for Ubuntu from `kairos-ubuntu-24.04-core-amd64-generic-v3.5.3.iso` to `kairos-ubuntu-24.04-core-amd64-generic-v3.5.6.iso`
- Updated Alpine base image from `alpine:3.21` to `alpine:3.22`

These changes ensure compatibility with the latest versions and improve the overall stability of the builds.

Signed-off-by: Itxaka <itxaka@kairos.io>

Itxaka committed:
feat: Update dependencies in go.mod and go.sum

- Updated `github.com/quic-go/quic-go` from v0.54.0 to v0.55.0
- Updated various `golang.org/x` dependencies to their latest versions:
  - `golang.org/x/crypto` from v0.41.0 to v0.43.0
  - `golang.org/x/exp` from v0.0.0-20250606033433 to v0.0.0-20251023183803
  - `golang.org/x/image` from v0.25.0 to v0.32.0
  - `golang.org/x/mod` from v0.27.0 to v0.29.0
  - `golang.org/x/net` from v0.43.0 to v0.46.0
  - `golang.org/x/sync` from v0.16.0 to v0.17.0
  - `golang.org/x/sys` from v0.35.0 to v0.37.0
  - `golang.org/x/text` from v0.28.0 to v0.30.0
  - `golang.org/x/tools` from v0.36.0 to v0.38.0

These updates are necessary to ensure compatibility with the latest features and security improvements in the dependencies.

Signed-off-by: Itxaka <itxaka@kairos.io>

Itxaka committed:
feat: Update go.mod and go.sum with new telemetry dependency

- Added `golang.org/x/telemetry` as an indirect dependency in go.mod
- Updated go.sum to reflect changes in dependencies

Signed-off-by: Itxaka <itxaka@kairos.io>

Itxaka committed:
Downgrade to alpine 3.21 for now

Some issues with encryption

Signed-off-by: Itxaka <itxaka@kairos.io>
